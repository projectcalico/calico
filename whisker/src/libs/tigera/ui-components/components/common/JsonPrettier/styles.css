.labelStyles {
    font-weight: normal;
    color: var(--chakra-colors-tigeraBlack);
}

.stringStyles {
    font-weight: normal;
    color: #dd0a73; /* Darker shade of var(--chakra-colors-tigeraRose); */
    padding-left: 4px;
}

.valueStyles {
    font-weight: normal;
    color: var(--chakra-colors-tigeraBlueMedium);
    padding-left: 4px;
}

.containerStyles {
    color: var(--chakra-colors-tigeraBlack);
    background-color: var(--chakra-colors-tigeraGrey-50);
    border: 1px solid;
    border-color: var(--chakra-colors-tigeraGrey-300);
    border-radius: 5px;
}
