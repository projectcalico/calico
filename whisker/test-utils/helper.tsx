import React, { ReactElement } from 'react';
import { render, renderHook, RenderOptions } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { theme } from '../src/theme';
import { MemoryRouter } from 'react-router';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            retry: false,
        },
    },
});

const queryClientProvider = ({ children }: any) => (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

export const renderHookWithQueryClient = (hook: () => unknown) =>
    renderHook(hook, { wrapper: queryClientProvider });

const Providers = ({ children }: { children: React.ReactNode }) => {
    return <ChakraProvider theme={theme}>{children}</ChakraProvider>;
};

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Providers, ...options });

export const renderWithRouter = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'> & {
        routes: string[];
    },
) =>
    customRender(
        <MemoryRouter initialEntries={options?.routes}>{ui}</MemoryRouter>,
        options,
    );

export * from '@testing-library/react';

export { customRender as render };
