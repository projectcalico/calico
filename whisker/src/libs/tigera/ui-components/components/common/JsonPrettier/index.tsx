import React from 'react';
import { JsonView, allExpanded, defaultStyles } from 'react-json-view-lite';
import 'react-json-view-lite/dist/index.css';
import './styles.css';

interface JsonPrettierProps {
    data: any;
    style?: Record<string, string>;
}
const JsonPrettier: React.FC<JsonPrettierProps> = ({ data, style }) => {
    const styles = {
        ...defaultStyles,
        container: 'containerStyles',
        label: 'labelStyles',
        stringValue: 'stringStyles',
        nullValue: 'valueStyles',
        numberValue: 'valueStyles',
        undefinedValue: 'valueStyles',
        ...style,
    };

    return (
        <JsonView data={data} shouldExpandNode={allExpanded} style={styles} />
    );
};

export default JsonPrettier;
