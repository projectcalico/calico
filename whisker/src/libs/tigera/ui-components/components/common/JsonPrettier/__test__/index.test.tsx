import { render } from '@/test-utils/helper';
import JsonPrettier from '..';

const defaultProps = {
    data: {
        a: 'one',
        b: 2,
        c: null,
        d: undefined,
        e: { f: 'g' },
        x: ['y', 'z'],
    },
};

describe('JsonPrettier', () => {
    it('should render', () => {
        const { asFragment } = render(<JsonPrettier {...defaultProps} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
