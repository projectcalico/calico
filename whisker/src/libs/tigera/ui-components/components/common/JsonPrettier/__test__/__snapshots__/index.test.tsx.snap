// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonPrettier should render 1`] = `
<DocumentFragment>
  <div
    class="containerStyles"
  >
    <div
      class="_2bkNM"
      role="list"
    >
      <span
        aria-controls=":jsnvw:1:"
        aria-expanded="true"
        aria-label="collapse JSON"
        class="_oLqym _f10Tu _1MFti _1LId0"
        role="button"
        tabindex="0"
      />
      <span
        class="_3uHL6 _3eOF8"
      >
        {
      </span>
      <div
        id=":jsnvw:1:"
      >
        <div
          class="_2bkNM"
          role="listitem"
        >
          <span
            class="labelStyles"
          >
            a:
          </span>
          <span
            class="stringStyles"
          >
            "one"
          </span>
          <span
            class="_3uHL6 _3eOF8"
          >
            ,
          </span>
        </div>
        <div
          class="_2bkNM"
          role="listitem"
        >
          <span
            class="labelStyles"
          >
            b:
          </span>
          <span
            class="valueStyles"
          >
            2
          </span>
          <span
            class="_3uHL6 _3eOF8"
          >
            ,
          </span>
        </div>
        <div
          class="_2bkNM"
          role="listitem"
        >
          <span
            class="labelStyles"
          >
            c:
          </span>
          <span
            class="valueStyles"
          >
            null
          </span>
          <span
            class="_3uHL6 _3eOF8"
          >
            ,
          </span>
        </div>
        <div
          class="_2bkNM"
          role="listitem"
        >
          <span
            class="labelStyles"
          >
            d:
          </span>
          <span
            class="valueStyles"
          >
            undefined
          </span>
          <span
            class="_3uHL6 _3eOF8"
          >
            ,
          </span>
        </div>
        <div
          class="_2bkNM"
          role="list"
        >
          <span
            aria-controls=":jsnvw:2:"
            aria-expanded="true"
            aria-label="collapse JSON"
            class="_oLqym _f10Tu _1MFti _1LId0"
            role="button"
            tabindex="0"
          />
          <span
            class="labelStyles"
          >
            e:
          </span>
          <span
            class="_3uHL6 _3eOF8"
          >
            {
          </span>
          <div
            id=":jsnvw:2:"
          >
            <div
              class="_2bkNM"
              role="listitem"
            >
              <span
                class="labelStyles"
              >
                f:
              </span>
              <span
                class="stringStyles"
              >
                "g"
              </span>
            </div>
          </div>
          <span
            class="_3uHL6 _3eOF8"
          >
            }
          </span>
          <span
            class="_3uHL6 _3eOF8"
          >
            ,
          </span>
        </div>
        <div
          class="_2bkNM"
          role="list"
        >
          <span
            aria-controls=":jsnvw:3:"
            aria-expanded="true"
            aria-label="collapse JSON"
            class="_oLqym _f10Tu _1MFti _1LId0"
            role="button"
            tabindex="0"
          />
          <span
            class="labelStyles"
          >
            x:
          </span>
          <span
            class="_3uHL6 _3eOF8"
          >
            [
          </span>
          <div
            id=":jsnvw:3:"
          >
            <div
              class="_2bkNM"
              role="listitem"
            >
              <span
                class="stringStyles"
              >
                "y"
              </span>
              <span
                class="_3uHL6 _3eOF8"
              >
                ,
              </span>
            </div>
            <div
              class="_2bkNM"
              role="listitem"
            >
              <span
                class="stringStyles"
              >
                "z"
              </span>
            </div>
          </div>
          <span
            class="_3uHL6 _3eOF8"
          >
            ]
          </span>
        </div>
      </div>
      <span
        class="_3uHL6 _3eOF8"
      >
        }
      </span>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
